### ISSUE TITLE
Finer classification of mesh connectivity requirements

### ISSUE THREAD

**Issue by boris-martin** [precice/precice#1384]
**Please describe the problem you are trying to solve.**

Connectivity requirement is currently internally defined as either "Vertices" or "Full" (meaning "connectivity is required). When not configured to Full, API calls to set up connectivity (like `setMeshTriangle`) are discarded. 
A finer classification should be used: if only edges are needed, we'd want to discard triangle/tetrahedra calls. (Although `setMeshTetrahedra` does set up edges at the moment. Which complicates things)

**Describe the solution you propose.**
Old draft at https://github.com/precice/precice/pull/1203
Basically, store the highest (in dimension) type of primitive required.

**Describe alternatives you've considered**
Deduce from interface dimension and coupling kind

**Comment by fsimonis** [comment]
Be simplifying the API #1322 we decided to preprocess connectivity of all Meshes in initialize.
Explicit connectivity may be rejected, but could lead of required implicit connectivity.
For the sake of simplicity, we keep the coarse On/Off mesh requirements.
